src/app/pages/services/components/service-webdav/service-webdav.component.ts

```diff
@@ -1,180 +1,185 @@
-  private webdav_certssl: FormSelectConfig;
-  private webdav_htauth: FormControl;
-  private webdav_password: FieldConfig;
-  private webdav_password2: FieldConfig;
-  private entityForm: EntityFormComponent;
+export class ServiceWebdavComponent implements OnInit {
+  isFormLoading = false;
+  pwdRequired = true;
+
+  form = this.fb.group({
+    protocol: [null as WebdavProtocol],
+    tcpport: [null as number, [Validators.min(1), Validators.max(65535)]],
+    tcpportssl: [null as number, [Validators.min(1), Validators.max(65535)]],
+    certssl: [null as number | string],
+    htauth: [''],
+    password: [''],
+    password2: ['', [this.validationService.matchOtherValidator('password')]],
+  });
+
+  readonly helptext = helptext;
+
+  protocol = {
+    fcName: 'protocol',
+    label: helptext.protocol_placeholder,
+    tooltip: helptext.protocol_tooltip,
+    options: of(helptext.protocol_options),
+  };
+
+  tcpport = {
+    fcName: 'tcpport',
+    label: helptext.tcpport_placeholder,
+    tooltip: helptext.tcpport_tooltip,
+    hidden: false,
+  };
+
+  tcpportssl = {
+    fcName: 'tcpportssl',
+    label: helptext.tcpportssl_placeholder,
+    tooltip: helptext.tcpportssl_tooltip,
+    hidden: false,
+  };
+
+  certssl = {
+    fcName: 'certssl',
+    label: helptext.certssl_placeholder,
+    tooltip: helptext.certssl_tooltip,
+    options: this.systemGeneralService.getCertificates().pipe(
+      map((certificates) => {
+        const options = certificates.map((certificate) => ({
+          label: certificate.name,
+          value: certificate.id,
+        }));
+
+        return [
+          { label: '---', value: null },
+          ...options,
+        ];
+      }),
+    ),
+    hidden: false,
+  };
+
+  htauth = {
+    fcName: 'htauth',
+    label: helptext.htauth_placeholder,
+    tooltip: helptext.htauth_tooltip,
+    options: of(helptext.htauth_options),
+  };
+
+  password = {
+    fcName: 'password',
+    label: helptext.password_placeholder,
+    tooltip: helptext.password_tooltip,
+  };
+
+  password2 = {
+    fcName: 'password2',
+    label: helptext.password2_placeholder,
+  };
 
   constructor(
+    private fb: FormBuilder,
     protected router: Router,
     protected route: ActivatedRoute,
     protected ws: WebSocketService,
-    protected _injector: Injector,
-    protected _appRef: ApplicationRef,
     protected systemGeneralService: SystemGeneralService,
     protected validationService: ValidationService,
+    private cdr: ChangeDetectorRef,
+    private dialogService: DialogService,
+    private errorHandler: FormErrorHandlerService,
   ) {}
 
-  resourceTransformIncomingRestData(data: WebdavConfig): any {
-    const transformed = { ...data };
-    const certificate = data['certssl'];
-    if (certificate && certificate.id) {
-      transformed['certssl'] = certificate.id;
-    }
-    delete transformed['password'];
-    return transformed;
-  }
+  ngOnInit(): void {
+    this.isFormLoading = true;
```

Comment:
Doesn't look like you are showing loading indicator in the form.

Reply:
![image](https://user-images.githubusercontent.com/22980553/140312207-5d46df29-e2f1-43b0-bbef-aa06bfed0888.png)

Reply:
Fixed.